packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/PreviewHostApiImpl.java

```diff
@@ -0,0 +1,142 @@
+    Preview.Builder previewBuilder = cameraXProxy.createPreviewBuilder();
+    if (rotation != null) {
+      previewBuilder.setTargetRotation(rotation.intValue());
+    }
+    if (targetResolution != null) {
+      previewBuilder.setTargetResolution(
+          new Size(
+              targetResolution.getWidth().intValue(), targetResolution.getHeight().intValue()));
+    }
+    Preview preview = previewBuilder.build();
+    instanceManager.addDartCreatedInstance(preview, identifier);
+  }
+
+  /**
+   * Sets the {@link Preview.SurfaceProvider} that will be used to provide a {@code Surface} backed
+   * by a Flutter {@link TextureRegistry.SurfaceTextureEntry} used to build the {@link Preview}.
+   */
+  @Override
+  public Long setSurfaceProvider(@NonNull Long identifier) {
+    Preview preview = (Preview) instanceManager.getInstance(identifier);
+    flutterSurfaceTexture = textureRegistry.createSurfaceTexture();
+    SurfaceTexture surfaceTexture = flutterSurfaceTexture.surfaceTexture();
+    Preview.SurfaceProvider surfaceProvider =
+        new Preview.SurfaceProvider() {
+          @Override
+          public void onSurfaceRequested(SurfaceRequest request) {
+            surfaceTexture.setDefaultBufferSize(
+                request.getResolution().getWidth(), request.getResolution().getHeight());
+            Surface flutterSurface = cameraXProxy.createSurface(surfaceTexture);
+            request.provideSurface(
+                flutterSurface,
+                Executors.newSingleThreadExecutor(),
+                (result) -> {
+                  SystemServicesFlutterApiImpl systemServicesFlutterApi =
+                      cameraXProxy.createSystemServicesFlutterApi(binaryMessenger);
+                  int resultCode = result.getResultCode();
+                  switch (resultCode) {
+                    case SurfaceRequest.Result.RESULT_SURFACE_USED_SUCCESSFULLY:
+                      flutterSurface.release();
+                      break;
+                    case SurfaceRequest.Result.RESULT_REQUEST_CANCELLED:
+                    case SurfaceRequest.Result.RESULT_INVALID_SURFACE:
+                    case SurfaceRequest.Result.RESULT_WILL_NOT_PROVIDE_SURFACE:
+                      flutterSurface.release();
+                    case SurfaceRequest.Result.RESULT_SURFACE_ALREADY_PROVIDED:
+                    default:
+                      systemServicesFlutterApi.sendCameraError(
+                          getProvideSurfaceErrorDescription(resultCode), reply -> {});
+                      break;
+                  }
+                });
+          };
+        };
+    preview.setSurfaceProvider(surfaceProvider);
+    return flutterSurfaceTexture.id();
+  }
+
+  /**
+   * Returns an error description for each {@link SurfaceRequest.Result} that represents an error
+   * with providing a surface.
+   */
+  private String getProvideSurfaceErrorDescription(int resultCode) {
```

Comment:
Is this a user facing string or a developer facing string or a string that is logged?